src/detect.c

```diff
@@ -1296,6 +1296,46 @@ static inline void StoreDetectFlags(DetectTransaction *tx, const uint8_t flow_fl
     }
 }
 
+static inline RuleMatchCandidateMergeSorted(DetectEngineThreadCtx *det_ctx, uint32_t j, uint32_t k)
+{
+    for (uint32_t i = det_ctx->match_array_cnt; i > 0;) {
+        const Signature *s = det_ctx->match_array[i - 1];
+        if (s->app_inspect != NULL) {
+            const SigIntId id = s->num;
+            if (j > 0) {
+                const RuleMatchCandidateTx *s0 = &det_ctx->tx_candidates[j - 1];
+                if (s->id > s0->id) {
```

Comment:
Shouldn't this comparison be with `num` instead of `id`, since `num` is what's used for sorting?

Reply:
`id` is correct. I do think the names of objects should probably have been consistent in these structs.

Reply:
Thinking about renaming `Signature::num` to `Signature::iid` (internal id)

Reply:
cf `DetectRunTxSortHelper`

Reply:
Nice catch Juliana